slides/workshop_rap_for_managers_basics.qmd

```diff
@@ -0,0 +1,343 @@
+
+> -   build extra time into projects to adopt new skills and practices where appropriate
+> -   learn the skills they need to manage software
+> -   evaluate RAP projects within organisations to understand and demonstrate the benefits of RAP
+> -   mandate their teams use RAP principles whenever possible
+
+# What does it actually mean for managers?
+
+# A nightmare scenario...
+
+-   You get a call from your team member, they're too ill to come to work this week
+-   They were the only one who knew how to run the code for your publication
+-   You're already behind schedule
+
+No problem! You can just run the code yourself, right?
+
+# Ok, first I need to find the code...
+
+::: {.column width="100%"}
+:::
+
+::: {.column width="40%"}
+This should be easy, it's in our shared folder because we haven't had a chance to move it to Azure DevOps yet...
+:::
+
+::: {.column width="60%"}
+![](images/code_nightmare_1.png){fig-align="center" width="60%"}
+:::
+
+# 
+
+::: {.column width="100%"}
+![](images/one-hour-later.jpg){fig-align="center" width="60%"}
+:::
+
+# Ok, I think I've found the right code...
+
+::: {.column width="100%"}
+:::
+
+::: {.column width="40%"}
+It's been a little while since I've run this code, I ought to check the documentation to make sure I do it right
+:::
+
+::: {.column width="60%"}
+![](images/code_nightmare_2.png){fig-align="center" width="75%"}
+:::
+
+# I'm sure there'll be comments in the code that tell me what to do...
+
+::: {.column width="100%"}
+:::
+
+::: {.column width="100%"}
+![](images/code_nightmare_3.png){fig-align="center" width="50%"}
+:::
+
+# Now there's a weird error
+
+::: {.column width="100%"}
+Time to ask for help!
+:::
+
+::: {.column width="100%"}
+:::
+
+::: {.column width="100%"}
+![](images/code_nightmare_4.jpg){fig-align="center" width="40%"}
+:::
+
+# Of course...
+
+-   This scenario is exaggerated, but it's not far from the truth of what we see in many teams
+-   It's not just about the time it takes to find the code and the stress it might cause, but the risk of errors and the lack of transparency
+-   Imagine how it would feel to encounter this issue as a new starter in the team
+
+# How could RAP have avoided these issues?
+
+# Finding the correct version of the code
+
+-   Sensible folder and file structure <span style="color:#4472c4;">(**Good practice**)</span>
+-   Version controlled final code scripts <span style="color:#70ad47;">(**Great practice**)</span>
+-   Collaboratively develop code using Git <span style="color:#f67400;">(**Best practice**)</span>
+
+Using Azure DevOps to store and version control your code scripts allows you to:
+
+-   Immediately see the current version of any code your team has been using
+-   View all historical versions of the code, along with when the changes were made, what they were, and who made them
+-   Roll back to previous versions of the code if you encounter an issue with the current version
+
```

Comment:
+ run each others versions of local code to debug / collaborate? Helpful when we come in to help too?